### ISSUE TITLE
Integer division done before cast to float giving wrong results

### ISSUE THREAD

**Issue by FunnyAnimal** [harold65/SolarMeter#3]
https://github.com/harold65/SolarMeter/blob/8f2cf97f13b269b32c362e6a9be9c3d1aeaded7b/PVoutput.ino#L46

Newer Arduino compilers (i use 1.8.3) do the integer division first, then cast to float. So e.g. when S[i]->Actual = 105 and S[i]->Factor = 10, the result becomes 10.0, not 10.5 . Better to code like this:

    float actual = (float)S[i]->Actual / S[i]->Factor;

Same issue for the following 2 lines.

**Comment by wspronk** [comment]
Same issue for the following 2 lines.
I do not see any other lines....

**Comment by FunnyAnimal** [comment]
This is line 46:
```
float actual = S[i]->Actual / S[i]->Factor;
```
And these are the following 2 lines:
```
float peak = S[i]->Peak / S[i]->Factor;
float today = S[i]->Today / S[i]->Factor;
```
All three lines first perform an "integer" division (‘‘truncation toward zero’’; see also https://stackoverflow.com/questions/3602827/what-is-the-behavior-of-integer-division), then cast the result to float. Which is incorrect, a "float" division should be done.

**Comment by harold65** [comment]
Will be implemented in 11.44
